sklearn/svm/_classes.py

```diff
@@ -877,6 +882,16 @@ def __init__(
         break_ties=False,
         random_state=None,
     ):
+        if probability != "deprecated":
```

Comment:
This should be moved to the `fit` method. As per the scikit-learn estimator API, the constructor parameters should always be stored unchanged as attributes. Any validation logic should be deferred to the `fit` method.